parsing/location.ml

```diff
@@ -546,6 +547,22 @@ let lines_around_from_lexbuf
     lines_around ~start_pos ~end_pos ~seek ~read_char
   end
 
+(* Try to get lines from a phrase buffer fix error #7925 *)
+let lines_around_from_phrasebuf
+    ~(start_pos: position) ~(end_pos: position)
+    (pb: Buffer.t):
+  input_line list
+  =
+  let pos = ref 0 in (* relative position (always starts at 0)*)
+  let seek n = pos := n in
+  let read_char () =
+    if !pos >= Buffer.length pb then (* end of buffer *) None
```

Comment:
Same here, handling `Buffer`s is straightforward enough that you can remove the comment.